api/services/user.go

```diff
@@ -37,24 +41,33 @@ func (s *service) UpdateUser(ctx context.Context, req *requests.UpdateUser) ([]s
 		return conflicts, NewErrUserDuplicated(conflicts, nil)
 	}
 
-	changes := &models.UserChanges{
-		Name:          req.Name,
-		Username:      strings.ToLower(req.Username),
-		Email:         strings.ToLower(req.Email),
-		RecoveryEmail: strings.ToLower(req.RecoveryEmail),
+	if req.Name != "" {
+		user.Name = cases.Title(language.AmericanEnglish).String(strings.ToLower(req.Name))
+	}
+
+	if req.Username != "" {
+		user.Username = strings.ToLower(req.Username)
+	}
+
+	if req.Email != "" {
+		user.Email = strings.ToLower(req.Email)
+	}
+
+	if req.RecoveryEmail != "" {
+		user.Preferences.SecurityEmail = strings.ToLower(req.RecoveryEmail)
```

Comment:
I think here we can use the previous strategy, which is always to the lowercase function call. The checking seems to be not worth and it complicates the code without a real, probably, good performance gain.